Sources/TokamakGTK/Shapes/Shape.swift

```diff
@@ -0,0 +1,233 @@
+//
+//  File.swift
+//  
+//
+//  Created by Morten Bek Ditlevsen on 29/12/2020.
+//
+
+import CGTK
+import CGDK
+import TokamakCore
+
+protocol ShapeAttributes {
+}
+
+extension _StrokedShape: ShapeAttributes {
+}
+
+//extension _ShapeView: ViewDeferredToRenderer {
+//  public var deferredBody: AnyView {
+//    let path = shape.path(in: .zero).deferredBody
+//    if let shapeAttributes = shape as? ShapeAttributes {
+//      return AnyView(HTML("div", shapeAttributes.attributes(style)) { path })
+//    } else if let color = style as? Color {
+//      return AnyView(HTML("div", [
+//        "style": "fill: \(color.cssValue(environment));",
+//      ]) { path })
+//    } else if let foregroundColor = foregroundColor {
+//      return AnyView(HTML("div", [
+//        "style": "fill: \(foregroundColor.cssValue(environment));",
+//      ]) { path })
+//    } else {
+//      return path
+//    }
+//  }
+//}
+
+extension _ShapeView: ViewDeferredToRenderer {
+  public var deferredBody: AnyView {
+    dump(self.environment)
+
+    return AnyView(WidgetView(build: { _ in
+      let w = gtk_drawing_area_new()
+      bindAction(to: w!)
+      return w!
+    }) {
+    })
+  }
+
+  func bindAction(to drawingArea: UnsafeMutablePointer<GtkWidget>) {
+    drawingArea.connect2(signal: "draw", closure: { widget, cr in
+//      dump(widget)
+//      dump(cr)
+
+      let width = gtk_widget_get_allocated_width (widget)
+      let height = gtk_widget_get_allocated_height (widget)
+//      print("W", width, "H", height)
+
+      let c = (style as? Color) ?? foregroundColor ?? Color.black
+//      dump(c)
+
+      // XXX TODO, don't know why my environment doesn't work.
+      var environment = EnvironmentValues()
+      environment[_ColorSchemeKey] = .light
+
+      dump(environment)
+      let rgba = _ColorProxy(c).resolve(in: environment)
```

Comment:
I got a crash trying to resolve the color in the environment of the _ShapeView (since that environment is empty), so I hacked around it. 
I am a bit uncertain about why I am not getting the 'actual' environment with the color scheme already set, but getting an empty one instead...

Reply:
I debugged it so far to conclude that it only happens when applying stroke. So something about environment updating of a _ShapeView inside a _StrokedView?
I believe this also happens in the HTML/DOM version. At least I get a blank page when trying to stroke a shape.
I haven't really grokked all the ways an environment can be set and updated yet, so I am a bit lost here. :-/

Reply:
@MaxDesiatov Perhaps I could remove my workaround and create an issue for the bug - since (as far as I can tell) the issue is not related to TokamakGTK or this PR?

Reply:
Yes please, that makes perfect sense 👍

Reply:
It turns out that the issue is already represented as: https://github.com/TokamakUI/Tokamak/issues/322
I'll add a bit of extra information to the issue.